03-task-manager/starter/error-handler.js

```diff
@@ -0,0 +1,9 @@
+// for manage errors
+const errorHandlerMiddleware = (err, req, res, next) => {
+  console.error(err); // for checking whats going on
+  const statusCode = err.statusCode || 500;
+  const message = err.message || "Something wrong";
+  res.status(statusCode).json({ msg: message });
```

Comment:
in the video, he ends up creating a subclass of Error that he uses for the 404s in the controller. And then here uses `instanceof` to differentiate between his 404 errors and errors from mongoose.

Yours ultimately achieves the same result here